src/groupeddataframe/splitapplycombine.jl

```diff
@@ -1447,187 +1440,57 @@ function _combine_tables_with_first!(first::Union{AbstractDataFrame,
+"""
+select(gd::GroupedDataFrame, args...;
+       copycols::Bool=true, keepkeys::Bool=true, regroup::Bool=false) =
+    _combine_executor(gd, args..., copycols=copycols, keepkeys=keepkeys,
+                      regroup=regroup, keeprows=true)
 
-julia> by(df, :a) do sdf # dropping group when DataFrame() is returned
-          sdf.c[1] != 1 ? sdf : DataFrame()
-       end
-6×3 DataFrame
-│ Row │ a     │ b     │ c     │
-│     │ Int64 │ Int64 │ Int64 │
-├─────┼───────┼───────┼───────┤
-│ 1   │ 2     │ 1     │ 2     │
-│ 2   │ 2     │ 1     │ 6     │
-│ 3   │ 3     │ 2     │ 3     │
-│ 4   │ 3     │ 2     │ 7     │
-│ 5   │ 4     │ 1     │ 4     │
-│ 6   │ 4     │ 1     │ 8     │
+"""
+    transform(gd::GroupedDataFrame, args...;
+              copycols::Bool=true, keepkeys::Bool=true, regroup::Bool=false)
 
-julia> by(df, :a, :b => :b1, :c => :c1,
-               [:b, :c] => +, keepkeys=false) # auto-splatting, renaming and keepkeys
-8×3 DataFrame
-│ Row │ b1    │ c1    │ b_c_+ │
-│     │ Int64 │ Int64 │ Int64 │
-├─────┼───────┼───────┼───────┤
-│ 1   │ 2     │ 1     │ 3     │
-│ 2   │ 2     │ 5     │ 7     │
-│ 3   │ 1     │ 2     │ 3     │
-│ 4   │ 1     │ 6     │ 7     │
-│ 5   │ 2     │ 3     │ 5     │
-│ 6   │ 2     │ 7     │ 9     │
-│ 7   │ 1     │ 4     │ 5     │
-│ 8   │ 1     │ 8     │ 9     │
+An equivalent of
+`select(gd, :, args..., copycols=copycols, keepkeys=keepkeys, regroup=regroup)`
+"""
+transform(gd::GroupedDataFrame, args...;
+          copycols::Bool=true, keepkeys::Bool=true, regroup::Bool=false) =
+    select(gd, :, args..., copycols=copycols, keepkeys=keepkeys,
+           regroup=regroup)
 
-julia> by(df, :a, :b, :c => sum) # passing columns and broadcasting
-8×3 DataFrame
-│ Row │ a     │ b     │ c_sum │
-│     │ Int64 │ Int64 │ Int64 │
-├─────┼───────┼───────┼───────┤
-│ 1   │ 1     │ 2     │ 6     │
-│ 2   │ 1     │ 2     │ 6     │
-│ 3   │ 2     │ 1     │ 8     │
-│ 4   │ 2     │ 1     │ 8     │
-│ 5   │ 3     │ 2     │ 10    │
-│ 6   │ 3     │ 2     │ 10    │
-│ 7   │ 4     │ 1     │ 12    │
-│ 8   │ 4     │ 1     │ 12    │
+"""
+    select!(gd::GroupedDataFrame{DataFrame}, args...; regroup::Bool=false)
 
-julia> by(df, :a, [:b, :c] .=> Ref)
-4×3 DataFrame
-│ Row │ a     │ b_Ref    │ c_Ref    │
-│     │ Int64 │ SubArra… │ SubArra… │
-├─────┼───────┼──────────┼──────────┤
-│ 1   │ 1     │ [2, 2]   │ [1, 5]   │
-│ 2   │ 2     │ [1, 1]   │ [2, 6]   │
-│ 3   │ 3     │ [2, 2]   │ [3, 7]   │
-│ 4   │ 4     │ [1, 1]   │ [4, 8]   │
+An equivalent of
+`select(gd, args..., copycols=false, keepkeys=true, regroup=regroup)`
+but updates the `parent(gd)` in place.
+"""
+function select!(gd::GroupedDataFrame{DataFrame}, args...; regroup::Bool=false)
+    newdf = select(gd, args..., copycols=false, regroup=false)
+    df = parent(gd)
+    copy!(_columns(df), _columns(newdf))
+    x = index(df)
+    copy!(_names(x), _names(newdf))
+    empty!(x.lookup)
+    for (i, n) in enumerate(x.names)
+        x.lookup[n] = i
+    end
```

Comment:
Don't we have a function doing this somewhere? I think this pattern is used in several places.

Reply:
A good point. I have introduced `_replace_columns!` function which also simplified `select!` for `DataFrame`